### ISSUE TITLE
Ver 19 - Default Map not updating/drawing properly

### ISSUE THREAD

**Issue by djehmli** [jaszhix/NoMansConnect#61]
- Description of the problem:
Version 19.0
Locations are now saving properly and am able to teleport successfully.  Location map is not updating/syncing as efficiently.  When starting the application there is a five minute or more delay before points are drawn, even after forcing a manual sync.  Clicking on a location is not highlighting the associated location in the map.  When switching from one galaxy to another the map is either empty or reflects the previous galaxy for a length of time from one or two minutes to 15 minutes, where before they were almost instantaneous.

If I have the app running for a half hour or more, gradually it becomes more and more responsive with items displaying close to but not as fast as previous versions.  Restarting the application does not improve response time, makes it worse.

Not sure if I have a file which didn't get updated properly or not.  But as a fail safe I did a full reinstall with the same results.

Also note, still getting issue with the log indicating the install location is not found.  I've changed it twice from the explicit directory of Binaries, back to the generic No Man's Sky with same results.  For versions up to 17, the Binaries directory was the default and no errors regarding the install path had been logged.

Regardless of the error message, locations are saving and teleport able, but not sure if the install error could also be causing the map update delay.  (Note, the 3d map is actually more responsive)

- Windows 10
- C:\\Users\\djehm\\AppData\\Roaming\\HelloGames\\NMS\\st_76561198052970098
- C:\Program Files (x86)\Steam\steamapps\common\No Man's Sky\Binaries

Wed Oct 25 2017 20:08:30 GMT-0400 (Eastern Daylight Time):    "Initializing No Man's Connect 0.19.0"
Wed Oct 25 2017 20:08:30 GMT-0400 (Eastern Daylight Time):    "Failed to locate NMS install: path doesn't exist."
Wed Oct 25 2017 20:08:32 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:08:32 GMT-0400 (Eastern Daylight Time):    "Active save file: C:\\Users\\djehm\\AppData\\Roaming\\HelloGames\\NMS\\st_76561198052970098\\save.hg"
Wed Oct 25 2017 20:08:32 GMT-0400 (Eastern Daylight Time):    "Username: djehmli"
Wed Oct 25 2017 20:08:32 GMT-0400 (Eastern Daylight Time):    "Current location: -1521:5:-84:179:122:0"
Wed Oct 25 2017 20:12:34 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:12:49 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:15:25 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:15:39 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:20:50 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."
Wed Oct 25 2017 20:21:05 GMT-0400 (Eastern Daylight Time):    "Finished reading No Man's Sky v4616 save file."

**Comment by jaszhix** [comment]
I reproduced this bug as well, I will push a fix for it next.

The responsiveness is linked to how many cached locations you have enabled. I recommend renaming backing up the `cache.json` file in `C:\Users\djehm\AppData\Roaming\NoMansConnect` and running NMC with a clean location cache. For the install location, try pointing it to `C:\Program Files (x86)\Steam\steamapps\common\No Man's Sky\`. 

The 3D map is more responsive, but I found its almost unusable if there's more than 8000 locations cached. Performance improvements are still a work in progress. The app has additional worker threads added several versions ago which definitely help with CPU bound operations, but there's a lot more that can be refactored.
